verifier/src/lib.rs

```diff
@@ -40,20 +41,19 @@ pub fn verify(
     stack_outputs: StackOutputs,
     proof: ExecutionProof,
 ) -> Result<u32, VerificationError> {
+    // get security level of the proof
+    let security_level = proof.security_level();
+
     // build public inputs and try to verify the proof
     let pub_inputs = PublicInputs::new(program_info, stack_inputs, stack_outputs);
-    let (hasher, proof) = proof.into_inner();
-    let security_level = hasher.security_level();
-    match hasher {
-        HashFunction::Blake3_192 => {
-            winter_verifier::verify::<ProcessorAir, Blake3_192>(proof, pub_inputs)
-        }
-        HashFunction::Blake3_256 => {
-            winter_verifier::verify::<ProcessorAir, Blake3_256>(proof, pub_inputs)
-        }
-        HashFunction::Rpo256 => winter_verifier::verify::<ProcessorAir, Rpo256>(proof, pub_inputs),
+    let (hash_fn, proof) = proof.into_parts();
+    match hash_fn {
+        HashFunction::Blake3_192 => verify_proof::<ProcessorAir, Blake3_192>(proof, pub_inputs),
+        HashFunction::Blake3_256 => verify_proof::<ProcessorAir, Blake3_256>(proof, pub_inputs),
+        HashFunction::Rpo256 => verify_proof::<ProcessorAir, Rpo256>(proof, pub_inputs),
```

Comment:
If we removed rpo from the CLI, then maybe this should be unreachable?

Reply:
N/A, this is the verifier lib, not the CLI verifier

Reply:
Yeah - for now RPO can be used manually via Rust interfaces but not via CLI.